src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,170 @@
 package core.basesyntax;
 
+//import jdk.internal.util.ArraysSupport;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+
 public class ArrayList<T> implements List<T> {
+
+
+    private static final int def_capacity = 10;
+    private static final Object[] empty_arraydata = {};
+
+
+    private T[] array = (T[]) new Object[def_capacity];
+    private int size;
+    private int cursor;
+
+
+    private String errorMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+
+    private Object[] grow() {
+        int oldCapacity = array.length;
+        double multiplicator = 1;
+        if (oldCapacity == size) {
+            multiplicator = 1.5;
+        }
+        if (oldCapacity > 0 || array != empty_arraydata) {
+            int newCapacity = (int) (oldCapacity * multiplicator + 1);
+            return array = Arrays.copyOf(array, newCapacity);
+        } else {
+            return array = (T[]) new Object[def_capacity];
+        }
```

Comment:
Complicated if-else construction. See [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#dont-complicate-if-else-construction-detailed-explanation).

First of all, `array != empty_arraydata` should be removed. Next, you have to resize your internal array in case the length of your array is equal to the size of your ArrayList, so all array positions are filled with the existing elements. It may be as it: `array.length == size`.

Also, you can create a constant to store the resize factor, e.g:
`private static final int RESIZE_MULTIPLICATOR = 1.5`

So, applying all the related comments above, your code may be simplified to this:
```suggestion
    private void grow() {
        if (array.length == size) {
            array = Arrays.copyOf(array, size * RESIZE_MULTIPLICATOR);
        }
    }
```